addons/api/mirage/factories/host-set.js

```diff
@@ -42,22 +47,34 @@ export default factory.extend({
   },
 
   attributes() {
-    // AWS specific
-    if (this.plugin?.name === 'aws') {
-      const filtersAmount = faker.number.int({ min: 1, max: 5 });
-      let filters = [];
-      for (let i = 0; i < filtersAmount; ++i) {
-        filters.push(`${faker.word.words(1)}=${faker.word.words(1)}`);
+    switch (this.plugin?.name) {
+      // AWS specific
+      case TYPE_HOST_CATALOG_PLUGIN_AWS: {
+        const filtersAmount = faker.number.int({ min: 1, max: 5 });
+        let filters = [];
+        for (let i = 0; i < filtersAmount; ++i) {
+          filters.push(`${faker.word.words(1)}=${faker.word.words(1)}`);
+        }
+        return {
+          filters,
+        };
+      }
+      // Azure specific
+      case TYPE_HOST_CATALOG_PLUGIN_AZURE:
+        return {
+          filter: `${faker.database.column()}=${faker.database.collation()}`,
+        };
+      // GCP specific
+      case TYPE_HOST_CATALOG_PLUGIN_GCP: {
+        const filtersAmount = faker.number.int({ min: 1, max: 5 });
+        let filters = [];
+        for (let i = 0; i < filtersAmount; ++i) {
+          filters.push(`label.${faker.word.words(1)}=${faker.word.words(1)}`);
```

Comment:
nitpick- we could move this block to a helper function

Reply:
Good call, pulled it out